huaweicloud/services/dds/resource_huaweicloud_dds_audit_log_policy.go

```diff
@@ -0,0 +1,282 @@
+		getAuditLogPolicyProduct = "dds"
+	)
+	getAuditLogPolicyClient, err := cfg.NewServiceClient(getAuditLogPolicyProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDS Client: %s", err)
+	}
+
+	getAuditLogPolicyPath := getAuditLogPolicyClient.Endpoint + getAuditLogPolicyHttpUrl
+	getAuditLogPolicyPath = strings.ReplaceAll(getAuditLogPolicyPath, "{project_id}",
+		getAuditLogPolicyClient.ProjectID)
+	getAuditLogPolicyPath = strings.ReplaceAll(getAuditLogPolicyPath, "{instance_id}", d.Id())
+
+	getAuditLogPolicyOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+	}
+	getAuditLogPolicyResp, err := getAuditLogPolicyClient.Request("GET",
+		getAuditLogPolicyPath, &getAuditLogPolicyOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving DDS audit log")
+	}
+
+	getAuditLogPolicyRespBody, err := utils.FlattenResponse(getAuditLogPolicyResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	keepDays := utils.PathSearch("keep_days", getAuditLogPolicyRespBody, 0)
+	if keepDays.(float64) == 0 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("keep_days", utils.PathSearch("keep_days", getAuditLogPolicyRespBody, nil)),
+		d.Set("audit_scope", utils.PathSearch("audit_scope", getAuditLogPolicyRespBody, nil)),
+		d.Set("audit_types", utils.PathSearch("audit_types", getAuditLogPolicyRespBody, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceDdsAuditLogPolicyDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// deleteAuditLog: Delete DDS audit log
+	var (
+		deleteAuditLogPolicyHttpUrl = "v3/{project_id}/instances/{instance_id}/auditlog-policy"
+		deleteAuditLogPolicyProduct = "dds"
+	)
+	deleteAuditLogPolicyClient, err := cfg.NewServiceClient(deleteAuditLogPolicyProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDS Client: %s", err)
+	}
+
+	deleteAuditLogPolicyPath := deleteAuditLogPolicyClient.Endpoint + deleteAuditLogPolicyHttpUrl
+	deleteAuditLogPolicyPath = strings.ReplaceAll(deleteAuditLogPolicyPath, "{project_id}",
+		deleteAuditLogPolicyClient.ProjectID)
+	deleteAuditLogPolicyPath = strings.ReplaceAll(deleteAuditLogPolicyPath, "{instance_id}", d.Id())
+
+	deleteAuditLogPolicyOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	deleteAuditLogPolicyOpt.JSONBody = utils.RemoveNil(buildDeleteAuditLogPolicyBodyParams(d))
+	_, err = deleteAuditLogPolicyClient.Request("POST", deleteAuditLogPolicyPath, &deleteAuditLogPolicyOpt)
+	if err != nil {
+		return diag.Errorf("error deleting DDS audit log policy: %s", err)
+	}
+	d.SetId("")
```

Comment:
the statement is unnecessary, please delete it.

Reply:
done